docs/USAGE-GUIDE.md

```diff
@@ -1560,6 +1568,41 @@ handshake.
 
 **s2n_connection_is_session_resumed** returns 1 if the handshake was abbreviated, otherwise returns 0, for tls versions < TLS1.3.
 
+## Additional TLS1.3 Session Resumption Related Calls
+
+Session resumption works differently in versions TLS1.3 and higher. Session ticket messages are now sent after the handshake in "post-handshake" messages, and multiple session tickets may be issued for the same connection. While some of the TLS1.2 session resumption APIs are still relevant for TLS1.3 session resumption, additional APIs are needed to utilize all the capabilities of TLS1.3 session resumption.
+
+```c
+int s2n_config_set_initial_ticket_count(struct s2n_config *config, uint8_t num);
+int s2n_connection_add_new_tickets_to_send(struct s2n_connection *conn, uint8_t num);
+int s2n_connection_set_server_keying_material_lifetime(struct s2n_connection *conn, uint32_t lifetime_in_secs);
+
+typedef int (*s2n_session_ticket_fn)(struct s2n_connection *conn, void *ctx, struct s2n_session_ticket *ticket);
+int s2n_config_set_session_ticket_cb(struct s2n_config *config, s2n_session_ticket_fn callback, void *ctx);
+int s2n_session_ticket_get_data_len(struct s2n_session_ticket *ticket, size_t *data_len);
+int s2n_session_ticket_get_data(struct s2n_session_ticket *ticket, size_t max_data_len, uint8_t *data);
+int s2n_session_ticket_get_lifetime(struct s2n_session_ticket *ticket, uint32_t *session_lifetime);
+```
+
+**s2n_config_set_initial_ticket_count** sets the initial number of session tickets to send.
+
+**s2n_connection_add_new_tickets_to_send** increases the number of session tickets to send by **num**.
+
+**s2n_connection_set_server_keying_material_lifetime** sets the keying material lifetime for session tickets. Use this to ensure session tickets don't get reissued past the lifetime of the certificate used to authenticate the original full handshake.
+
+**s2n_session_ticket_fn** is invoked whenever a client receives a session ticket. Use this callback in conjunction with the **s2n_session_ticket** getters to get the serialized ticket data and related information. A **ctx** pointer is provided to let a user pass state to the callback, if needed.
+
+**s2n_config_set_session_ticket_cb** sets the session ticket callback function to be invoked whenever the client receives
```

Comment:
Can the malicious server just send tickets all the time potentially causing issues to the client if callback implementation is expensive or keeps allocating memory? If so should we note about the risks in documentaiton?

Reply:
Added a note about the risks. I think this is relatively low-risk, given that all of this is happening post-handshake, but clients should be aware the server can send a lot of tickets.